sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventDataBatch.java

```diff
@@ -182,6 +208,18 @@ private int getSize(final EventData eventData, final boolean isFirst) {
         Objects.requireNonNull(eventData, "'eventData' cannot be null.");
 
         final Message amqpMessage = createAmqpMessage(eventData, partitionKey);
+        if (isCreatedByIdempotentProducer) {
+            // Pre-allocate size for system properties "com.microsoft:producer-sequence-number".
+            // EventData doesn't have this system property until it's added just before an idempotent producer
+            // sends the EventData out.
+            final MessageAnnotations messageAnnotations = (amqpMessage.getMessageAnnotations() == null)
+                ? new MessageAnnotations(new HashMap<>())
+                : amqpMessage.getMessageAnnotations();
+            amqpMessage.setMessageAnnotations(messageAnnotations);
+            messageAnnotations.getValue().put(
+                Symbol.getSymbol(
+                    AmqpMessageConstant.PRODUCER_SEQUENCE_NUMBER_ANNOTATION_NAME.getValue()), Integer.MAX_VALUE);
+        }
         int eventSize = amqpMessage.encode(this.eventBytes, 0, maxMessageSize); // actual encoded bytes size
         eventSize += 16; // data section overhead
```

Comment:
Orthogonal to these changes, but this seems a bit different than how .NET does the estimation, which goes back to the T1.    Not sure whether it's a concern, just mentioning for awareness:
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpEventBatch.cs#L137

Reply:
Thanks for the feedback on this. I'll follow-up to ask